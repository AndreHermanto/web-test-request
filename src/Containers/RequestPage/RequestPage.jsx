import React, {Component} from 'react';
import {
  FormGroup,
  FormControl,
  ControlLabel
} from 'react-bootstrap';


/**
 * RequestPage - this generates the main page sending the test request.
 */
class RequestPage extends Component {
  constructor(props) {
    super(props); 
    
    this.state = {
      testList: []
    };
  }
  
  componentWillMount() {
    this.getList('#','testList'); // # is only temporary url
  }
  
  /**
   * A shared function to retrieve the option list from the gateway
   * e.g., retrieving the test list and gene list
   *
   * @param {String} url - url of the api call
   * @param {String} stateChild - name of the state that the list applies
   */
  getList(url, stateChild) {
    var func = this;
    fetch(url)
      .then(
        (response) => response.text()
      )
      .then(
        (json) => {
          func.setState(
            (state) => {
              state[stateChild] = json;
              return state;
            }
          );
        }
      );
  }
  
  /**
   * A shared function to generate options from a list contained in state.
   * e.g., generating options for the test list and gene list
   *
   * @param {String} stateChild - name of the state that the list applies
   */
  genOptions(stateChild) {
    this.state[stateChild].map(
      (item) => {
        return <option value={item}>{item}</option>
      }
    );
  }
  
  render() {
    return (
      
      <div>
        <h4>Web Test Request</h4>

        <FormGroup>
          <ControlLabel>Available Test List</ControlLabel>
          <FormControl 
            componentClass="select" 
            placeholder="Select test" 
            name="test"
          >
            
          {this.genOptions('testList')}
            
          </FormControl>
        </FormGroup>
      </div>
      
    );
  }
};

export default RequestPage;