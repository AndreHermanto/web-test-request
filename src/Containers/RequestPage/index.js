import React, {Component} from 'react';
import {
  FormGroup,
  FormControl,
  ControlLabel
} from 'react-bootstrap';

/**
 * RequestPage - this generates the main page sending the test request.
 */
function RequestPage(props) {
  return (

    <div>
      <h4>Web Test Request</h4>
      <FormGroup>
        <ControlLabel>Available Test List</ControlLabel>
        <FormControl 
          componentClass="select" 
          placeholder="Select test" 
          name="test"
        >

        {
          props.testList.map(
            (test) => {
              return <option key={test} value={test}>{test}</option>
            }
          )
        }

        </FormControl>
      </FormGroup>
    </div>

  );
}

export default RequestPage;