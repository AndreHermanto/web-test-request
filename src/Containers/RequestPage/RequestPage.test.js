import React from 'react';
import ReactDOM from 'react-dom';
import fetchMock from 'fetch-mock';
import renderer from 'react-test-renderer';
import RequestPage from './index';

describe('App testing', () => {
  it('RequestPage renders without crashing', () => {
    fetchMock.get('*', [{id:'1',label:'a'},{id:'2',label:'b'},{id:'3',label:'c'}]);
    const tree = renderer.create(<RequestPage />).toJSON();
    expect(tree).toMatchSnapshot();
    fetchMock.restore();
  });
});
