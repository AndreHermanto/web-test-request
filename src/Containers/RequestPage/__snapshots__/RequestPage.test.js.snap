exports[`App testing RequestPage renders without crashing 1`] = `
<div>
  <h4>
    Web Test Request
  </h4>
  <div
    className="form-group">
    <label
      className="control-label"
      htmlFor={undefined}>
      Available Test List
    </label>
    <select
      className="form-control"
      id={undefined}
      name="test"
      placeholder="Select test"
      type={undefined} />
  </div>
</div>
`;
